### DISCUSSION TITLE
Allure-jest. Error: Could not resolve a module for a custom reporter WHEN I run tests

### DISCUSSION THREAD

**Discussion by kemamoto** [allure-framework/allure2#2814] [Questions & Support] [answered]
Hi all,
I've got a problem with configuring allure-jest reporter.
As I know allure-jest and jest-allure are different things and it's preferable to use allure jest.
I installed allure-jest and configured jest.config.ts for my project where I use allure-jest as my reporter. When I run my test it fails with the error. Can you please help with that. When I comment the row with allure-jest reporter and leave only default reporter my tests are [assed successfully.
My package.json
```
{
  "name": "project_at",
  "version": "1.0.0",
  "scripts": {
    "test": "npx playwright test",
    "allure:report": "allure serve allure-results --host localhost --port 8000",
    "allure:clean": "rm -rf allure-results/*"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "description": "",
  "devDependencies": {
    "@babel/preset-env": "^7.26.0",
    "@babel/preset-typescript": "^7.26.0",
    "@types/jest": "^29.5.14",
    "@types/node": "^22.8.2",
    "allure-jest": "^3.0.6",
    "allure-js-commons": "^3.0.6",
    "allure-playwright": "^3.0.6",
    "commandline": "^0.0.2",
    "jest": "^29.7.0",
    "ts-jest": "^29.2.5",
    "ts-node": "^10.9.2",
    "typescript": "^5.6.3"
  },
  "dependencies": {
    "@playwright/test": "^1.48.2",
    "axios": "^1.7.7",
    "dotenv": "^16.4.5"
  }
}
```

And my jest.config.ts
```
import type { Config } from '@jest/types';

const config: Config.InitialOptions = {
  preset: 'ts-jest',
  testEnvironment: 'node',
  testMatch: ['<rootDir>/src/tests/api/**/*.test.ts'],
  reporters: [
      'default',
    ['allure-jest', { outputDirectory: 'allure-results' }],
  ],
  globals: {
    testTimeout: 50000,
  },
  verbose: true
};

export default config;
```

See the img with the error
![Screenshot at Nov 26 00-39-07](https://github.com/user-attachments/assets/3bc1a8d8-a10b-4c57-bc42-d57174b33118)

**Comment by baev** [comment, marked as answer]
Allure Jest shouldn't be configured as a reporter. It's a test environment. Please see documentation (step 7) https://allurereport.org/docs/jest/.

You can also find the working example here: https://github.com/allure-examples/jest-npm-esm-ts 

In your case, you should replace 

```
testEnvironment: 'node',
```

with 

```
testEnvironment: 'allure-jest/node',
```

and remove `['allure-jest', { outputDirectory: 'allure-results' }],`

**Reply by kemamoto** [reply]
Thanks a lot!
